lib/yargs-parser-types.ts

```diff
@@ -0,0 +1,159 @@
+import type { Dictionary, KeyOf, ValueOf } from './common-types'
+
+export type ArgsInput = string | any[];
+
+export type ArgsOutput = (string | number)[];
+
+export interface Arguments {
+  /** Non-option arguments */
+  _: ArgsOutput;
+  /** Arguments after the end-of-options flag `--` */
+  '--'?: ArgsOutput;
+  /** All remaining options */
+  [argName: string]: any;
+}
+
+export interface DetailedArguments {
+  /** An object representing the parsed value of `args` */
+  argv: Arguments;
+  /** Populated with an error object if an exception occurred during parsing. */
+  error: Error | null;
+  /** The inferred list of aliases built by combining lists in opts.alias. */
+  aliases: Dictionary<string[]>;
+  /** Any new aliases added via camel-case expansion. */
+  newAliases: Dictionary<boolean>;
+  /** Any new argument created by opts.default, no aliases included. */
+  defaulted: Dictionary<boolean>;
+  /** The configuration loaded from the yargs stanza in package.json. */
+  configuration: Configuration;
+}
+
+export interface Configuration {
+  /** Should variables prefixed with --no be treated as negations? Default is `true` */
+  'boolean-negation': boolean;
+  /** Should hyphenated arguments be expanded into camel-case aliases? Default is `true` */
```

Comment:
these configuration keys should all be optional I believe, when testing I got:

```bash
TS2580 [ERROR]: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
  process = process || {env: []}
```

Until I updated them to be optional.